### ISSUE TITLE
Android-x86 StandaloneVMs broken after Qubes R4.0 update

### ISSUE THREAD

**Issue by antarisherdeng4** [QubesOS/qubes-issues#7374]
### Qubes OS release
R4.0

### Brief summary
Today after Qubes update (last Qubes update was 1-1.5 months ago) I can't start any of my Android-x86 StandaloneVMs, I'm getting stuck at "Detecting Android-x86".
I've installed these VMs using @unman guide:
https://github.com/unman/notes/blob/master/InstallingAndroid.md
I thought that patched files got rewritten after Qubes update and run patch once again but it didn't fix the issue.
I've tried to create and install new Android-x86 StandaloneVM and it works fine.
My problem is how to get my old existing Android-x86 StandaloneVMs to work.

### Steps to reproduce
1. Have working Android-x86 StandaloneVM.
2. Update Qubes.

### Expected behavior
Existing Android-x86 StandaloneVMs boot and works fine.

### Actual behavior
Existing Android-x86 StandaloneVMs doesn't boot.

**Comment by marmarek** [comment]
Have you played with ReactOS in the past? This looks like you have https://github.com/unman/change_disk/blob/master/init_patch applied.

**Comment by antarisherdeng4** [comment]
I didn't play with ReactOS, but I did apply patch_init patch that call init_patch inside as stated in this guide:
https://github.com/unman/notes/blob/master/InstallingAndroid.md
I've looked into init_patch and it indeed check for the "react*" string in qube name and add some arguments but I don't understand how it works exactly. But everything worked before Qubes update even for VMs with name that don't start with "react".

**Comment by icequbes1** [comment]
Note that patching the stubdom rootfs likely doesn't qualify as a Qubes OS issue - this is probably better resolved on the forum or qubes-users. I've just tried an android-x86, patched the stubdom rootfs similar to what unman does, and had no issues.

According to unman's patch, your qube must start with the name "react". So, what you're observing is correct. Your earlier patch must have been different (e.g., not requiring the qube name to start with "react").

**Comment by antarisherdeng4** [comment]
> Note that patching the stubdom rootfs likely doesn't qualify as a Qubes OS issue - this is probably better resolved on the forum or qubes-users.

Yes it may be as you say so I've opened a new issue in @unman patch repo.

> I've just tried an android-x86, patched the stubdom rootfs similar to what unman does, and had no issues.

Did you try only for qube name starting with "react"?

>According to unman's patch, your qube must start with the name "react". So, what you're observing is correct. Your earlier patch must have been different (e.g., not requiring the qube name to start with "react").

No, it should work for every qube and not only for qube with name starting with "react". I've quoted in my previous post that the patch should work for all HVMs:

> As it stands, this change applies to **all** HVMs.

https://github.com/unman/notes/blob/master/disks_in_Qubes#L104

**Comment by andrewdavidwong** [comment]
@antarisherdeng4, is this still a problem on 4.1?

**Comment by unman** [comment]
That patch is no longer required *at all* in 4.1, so yes, this issue is
fixed.
